x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,295 @@
+import { i18n } from '@kbn/i18n';
+import { RadioGroup } from './csp_boxed_radio_group';
+import { getPosturePolicy, NewPackagePolicyPostureInput } from './utils';
+import { MIN_VERSION_GCP_CIS } from '../../common/constants';
+
+type SetupFormatGCP = 'google_cloud_shell' | 'manual';
+const GCPSetupInfoContent = () => (
+  <>
+    <EuiSpacer size="l" />
+    <EuiTitle size="s">
+      <h2>
+        <FormattedMessage
+          id="xpack.csp.gcpIntegration.setupInfoContentTitle"
+          defaultMessage="Setup Access"
+        />
+      </h2>
+    </EuiTitle>
+    <EuiSpacer size="l" />
+    <EuiText color={'subdued'} size="s">
+      <FormattedMessage
+        id="xpack.csp.gcpIntegration.setupInfoContent"
+        defaultMessage="The integration will need elevated access to run some CIS benchmark rules. Select your preferred
+    method of providing the GCP credentials this integration will use. You can follow these
+    step-by-step instructions to generate the necessary credentials."
+      />
+    </EuiText>
+  </>
+);
+
+/* NEED TO FIND THE REAL URL HERE LATER*/
+const DocsLink = (
+  <EuiText color={'subdued'} size="s">
+    <FormattedMessage
+      id="xpack.csp.gcpIntegration.docsLink"
+      defaultMessage="Read the {docs} for more details"
+      values={{
+        docs: (
+          <EuiLink href="https://cloud.google.com/docs/authentication" external>
+            documentation
+          </EuiLink>
+        ),
+      }}
+    />
+  </EuiText>
+);
+
+const CredentialFileText = i18n.translate(
+  'xpack.csp.findings.gcpIntegration.gcpInputText.credentialFileText',
+  { defaultMessage: 'Path to JSON file containing the credentials and key used to subscribe' }
+);
+const CredentialJSONText = i18n.translate(
+  'xpack.csp.findings.gcpIntegration.gcpInputText.credentialJSONText',
+  { defaultMessage: 'JSON blob containing the credentials and key used to subscribe' }
+);
+
+type GcpCredentialsType = 'credentials_file' | 'credentials_json';
+type GcpFields = Record<string, { label: string; type?: 'password' | 'text' }>;
+interface GcpInputFields {
+  fields: GcpFields;
+}
+
+const gcpField: GcpInputFields = {
+  fields: {
+    project_id: {
+      label: i18n.translate('xpack.csp.gcpIntegration.projectidFieldLabel', {
+        defaultMessage: 'Project ID',
+      }),
+      type: 'text',
+    },
+    credentials_file: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsFileFieldLabel', {
+        defaultMessage: 'Credentials File',
+      }),
+      type: 'text',
+    },
+    credentials_json: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsJSONFieldLabel', {
+        defaultMessage: 'Credentials JSON',
+      }),
+      type: 'text',
+    },
+  },
+};
+
+const credentialOptionsList = [
+  {
+    label: i18n.translate('xpack.csp.gcpIntegration.credentialsFileOption', {
+      defaultMessage: 'Credentials File',
+    }),
+    text: 'Credentials File',
+  },
```

Comment:
EUI's select component uses native html `option` element so it accepts label. but taking a look at [their implementation](https://github.com/elastic/eui/blob/0ccd0c7520b315129559faaa549118cd4603e85e/src/components/form/select/select.tsx) you'll see that `text` is being used for display, which is the [correct way of using this elements](https://www.w3schools.com/tags/tag_option.asp). 
Let's convert this select box to its optimal behaviour, using `text` for display (add the translations to text) and `value` to save data and control the form.

Reply:
so are you saying that we should remove labels and have text & value fields instead ?
and have the translation on text

Reply:
Yes exactly, sorry if wasn't clear. wanted to explain the reasoning behind it as well